File: java_with_DSA/linkedList.java


class linkedList{
    private Node head;
    private Node tail;
        private int  size;

    linkedList(){
        this.size=0;

    }
    public void  dispaly( ){
        Node temp=head;
        while (temp!=null) {
            System.out.print(temp.value+" --> "+temp.next+" -->  ");
            temp=temp.next;
        }

        System.out.println("end");

    }
public void inSertFirst(int val){
    Node node=new Node(val);
    node.next=head;
    head=node;
    System.out.println(node.value);
    System.out.println(node.next);
    

    if (tail==null) {
        tail=head;
        
    }
    size+=1;
}


    class Node{
        private   int value;
        private Node next;
        Node(int val ){
            this .value=val ;
        }
        Node(int val,Node next){
            this .value=val ;
            this .next=next ;

        }
        public static void main(String[] args) {
            linkedList ll=new linkedList();

                ll.inSertFirst(3
                )
                ;
                                ll.inSertFirst(3);  
                ll.inSertFirst(3);  
                ll.inSertFirst(3);  
                ll.inSertFirst(3);  
  ll.dispaly();
            }
    } 
}

File: DSA_with_javascript/linkedList.js
// linkedlist class
class LinkedList {
	constructor()
	{
		this.head = null;
		this.size = 0;
	}

	// functions to be implemented
	add(element)
	// insertAt(element, location)
	// removeFrom(location)
	// removeElement(element)

	// Helper Methods
	// isEmpty
	// size_Of_List
	// PrintList

// adds an element at the end
// of list
add(element)
{
	// creates a new node
	var node = new Node(element);

	// to store current node
	var current;

	// if list is Empty add the
	// element and make it head
	if (this.head == null)
		this.head = node;
	else {
		current = this.head;

		// iterate to the end of the
		// list
		while (current.next) {
			current = current.next;
		}

		// add node
		current.next = node;
	}
	this.size++;
}
}